pages/_app.tsx

```diff
@@ -25,9 +25,15 @@ export type MetaTagProps = {
   metaTags?: MetaTag[] | null;
 };
 
-const SafeHydrate: FC = ({ children }) => (
-  <div suppressHydrationWarning>{typeof window === 'undefined' ? null : children}</div>
-);
+function SafeHydrate({ children }: PropsWithChildren) {
+  const [render, setRender] = useState(false);
+
+  useEffect(() => {
+    setRender(true);
+  }, []);
+
+  return <div suppressHydrationWarning>{render ? children : null}</div>;
+}
```

Comment:
Achieves the same thing but this is the React approved way of doing it

Reply:
can we add a comment explaining this